app/src/main/java/org/oppia/app/parser/StringToFractionParser.kt

```diff
@@ -2,42 +2,106 @@ package org.oppia.app.parser
 
 import org.oppia.app.model.Fraction
 import org.oppia.domain.util.normalizeWhitespace
+import java.lang.Integer.parseInt
+import java.util.regex.Pattern
 
 /** This class contains method that helps to parse string to fraction. */
 class StringToFractionParser {
   private val wholeNumberOnlyRegex = """^-? ?(\d+)$""".toRegex()
   private val fractionOnlyRegex = """^-? ?(\d+) ?/ ?(\d)+$""".toRegex()
   private val mixedNumberRegex = """^-? ?(\d)+ ?(\d+) ?/ ?(\d)+$""".toRegex()
+  private val invalidCharsRegex = """^[\d\s/-]+$""".toRegex()
+  private val fractionRegex = """^\s*-?\s*((\d*\s*\d+\s*\/\s*\d+)|\d+)\s*$""".toRegex()
+
+  object FRACTION_PARSING_ERRORS {
+    const val INVALID_CHARS = "Please only use numerical digits, spaces or forward slashes (/)"
+    const val INVALID_FORMAT = "Please enter a valid fraction (e.g., 5/3 or 1 2/3)"
+    const val DIVISION_BY_ZERO = "Please do not put 0 in the denominator"
+  }
+
+  fun fromRawInputString(inputText: String): Fraction {
```

Comment:
This method is created as per oppia web... As per the suggestion of ben in yesterday's meeting... for validation... This can be removed if separate method is not required for validation and its only for approach confirmation.@ben please confirm...